reference: clarify support for foreign key contraints (#1591)
pingcap/docs: dev/reference/sql/constraints.md

- `PRIMARY KEY` and `UNIQUE` constraints are checked lazily by default. By batching checks until when the transaction commits, TiDB is able to reduce network communication. This behavior can be changed by setting `tidb_constraint_check_in_place` to `TRUE`.

- TiDB supports creating foreign key constraints, but these constraints are ineffective in DML statements.

## Foreign Key

> TiDB currently supports `FOREIGN KEY` constraints. For example:

Reviewer: I think "TiDB currently supports `FOREIGN KEY` creation in DDL commands.`" would be better, since we don't fully support it now.

Author: @wjhuang2016 Comments addressed. PTAL again, thanks.
Also I'll update the Chinese version in a new PR.

- `PRIMARY KEY` and `UNIQUE` constraints are checked lazily by default. By batching checks until when the transaction commits, TiDB is able to reduce network communication. This behavior can be changed by setting `tidb_constraint_check_in_place` to `TRUE`.

> - TiDB supports creating foreign key constraints, but these constraints are ineffective in DML statements.

Reviewer 2: ```suggestion
- TiDB supports creating foreign key constraints, but these constraints are not enforced by DML statements.
```
@wjhuang2016 Please take a look at this update suggested by @gregwebs .

Reviewer: LGTM
